app/models/goal.py

```diff
@@ -1,5 +1,40 @@
 from app import db
+from flask import make_response, jsonify, abort
+from datetime import datetime
 
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String, nullable=False)
+
+    tasks = db.relationship("Task", backref="goal", lazy=True)
+
+
+    def to_dict(self):
+        return {"goal": {
+            "goal_id": self.goal_id,
+            "title": self.title,
+            }
+        }
+
+
+    @classmethod
+    def from_dict(cls, data):
+        title = data.get("title", "")
+        return Goal(title=title)
```

Comment:
What happens if the class name is changed on line 6? We'd have to remember to change it here, too! Or! we can use the `cls` parameter:
```python
return cls(title=title)
```